mbuild/packing.py

```diff
@@ -355,9 +367,15 @@ def fill_region(
     # See if region is a single region or list
     if isinstance(region, Box):  # Cannot iterate over boxes
         region = [region]
+        if not bounds:
+            msg = (
+                "if passing in an mbuild.Box or list of mbuild.Box as `region`, "
+                "`bounds must also be defined as a list of the min and max xyz values"
+            )
+            raise ValueError(msg)
     elif not any(isinstance(reg, (list, set, Box)) for reg in region):
         region = [region]
-    region = [_validate_box(reg) for reg in region]
+    container = [_validate_box(bounding) for bounding in bounds]
```

Comment:
I'm a little confused on the logic here: it seems like region is not being used, so I don't know why we set it.

Reply:
I think it was my convoluted way to ensure that mins and maxs of the system are always passed. So far the assumption to fill a region is that it is still a rectangular box, and support for non-rectangular boxes are not yet available. Essentially this preserves current fill_region behavior, and a majority of this code will be refactored when packing.py is updated.

Reply:
I think I'm still confused why region isn't used in fill region. 🤔 `region` isn't accessed anywhere below line 377 in this function and instead bounds is used to define the container.

Reply:
Ive updated the logic, if a region is passed without a bounds, that region is used, if a bound is passed in with a region, that bound overrides it.

Although the current behavior is that if you pass in one bound, youre expected to pass them for each corresponding region. Finishing up this logic now, but its a bit closer now.